paddle/fluid/inference/tensorrt/plugin/emb_eltwise_layernorm_plugin.cu

```diff
@@ -31,11 +31,34 @@ namespace plugin {
 // Dynamic Plugin below.
 #if IS_TRT_VERSION_GE(6000)
 
-int EmbEltwiseLayernormPluginDynamic::initialize() {
+template <typename T>
+int EmbEltwiseLayernormPluginDynamic<T>::initialize() {
+  int nb_emb = embs_.size();
+  std::vector<void *> ptr_vector(nb_emb);
+  std::vector<std::vector<half>> emb_fp16(nb_emb);
+
+  if (sizeof(T) == sizeof(float)) {
+    // FP32
+    for (int i = 0; i < nb_emb; ++i) {
+      ptr_vector[i] = embs_[i];
+    }
+  } else {
+    // FP16
+    for (int i = 0; i < nb_emb; ++i) {
+      auto emb_size = emb_sizes_[i];
+      auto &tmp = emb_fp16[i];
+      tmp.resize(emb_size);
+
+      for (int j = 0; j < emb_size; ++j) {
+        tmp[j] = static_cast<half>(embs_[i][j]);
+      }
+      ptr_vector[i] = tmp.data();
+    }
+  }
   embs_gpu_.reserve(embs_.size());
```

Comment:
Should replace `embs_gpu_.reserve` with `embs_gpu_.resize`.

It is not safe to access the pointers of element of `embs_gpu_` before push anything into the vector.

This is a simple example. Try it and you will see that `bar` is reserved to 10, but size of `bar` still is 0.
```c++
#include <iostream>
#include <vector>

int main ()
{
  std::vector<int> bar;
  int N = 10;
  bar.reserve(N);

  for (int i=0; i<N; ++i) {
    int* ptr = &bar[i];
    std::cout << "iteration: " << i << std::endl;
    std::cout << "get &bar[i]: " << ptr << std::endl;
    std::cout << "bar.size: " << bar.size() << std::endl;
    std::cout << "bar.capacity: " << bar.capacity() << std::endl;
  }
  return 0;
}
```